File: src/stats/service/StatsService.kt
package stats.service

import ext.axios.Axios
import ext.axios.AxiosResponse
import stats.model.Summary
import kotlin.js.Promise

class StatsService {

    companion object {
        fun getSummary(): Promise<AxiosResponse<Summary>> = Axios.get("data/summary.json")
    }
}

File: src/stats/service/StatsService.ts
import Axios, {AxiosResponse} from 'axios';
import {Summary} from '../model/Summary';
import {DownloadSnapshot} from '../model/DownloadSnapshot';

class StatsService {
  getSummary(): Promise<AxiosResponse<Summary>> {
    return Axios.get('data/summary.json');
  }

  getDownloads(): Promise<AxiosResponse<DownloadSnapshot[]>> {
    return Axios.get('data/downloads.json');
  }
}

export default new StatsService();
